esp-wifi/src/lib.rs

```diff
@@ -388,8 +391,12 @@ pub fn init<'d, T: EspWifiTimerSource, R: EspWifiRngSource>(
 
     setup_radio_isr();
 
-    // This initializes the task switcher and timer tick interrupt.
-    preempt::setup(unsafe { timer.clone_unchecked() }.timer());
+    // This initializes the task switcher
+    preempt::enable();
+
+    // Enable timer tick interrupt
+    #[cfg(feature = "builtin-scheduler")]
+    preempt_builtin::setup_timer(unsafe { timer.clone_unchecked() }.timer());
```

Comment:
I think we need to setup the timer before enabling the scheduler, that should fix the errors?

Reply:
That does indeed seem to fix the error.

~~I don't understand why, and it seem dangerous. The timer interrupt does trigger a context switch, which would horribly fail if the context switching hasn't been set up.~~
Ah no, this is handled by disabling interrupts!